tests/test_collect.py

```diff
@@ -37,6 +37,28 @@ def task_write_text(depends_on, produces):
     assert tmp_path.joinpath("out.txt").read_text() == "Relative paths work."
 
 
+@pytest.mark.end_to_end()
+def test_collect_module_name_(tmp_path):
```

Comment:
I think we need a test that task modules can have the same name, eg `one/task_a.py` and `two/task_a.py`. I looked through this file (test_collect.py`) and I didn't see a test that did this, but maybe I'm missing it somewhere.

Reply:
Good idea! I guess this case is only indirectly covered by some other test and it is better to have it explicit. Added as `test_collect_tasks_from_modules_with_the_same_name`.